### PR TITLE
Create TestDataBuilder class

### PR DESCRIPTION
**Pull request by emmastephenson**
# BACKEND PULL REQUEST

## Related Issue

- Part one to #4462 

## Changes Proposed

- Creates a new class, TestDataBuilder
- Existing organization-related TestDataFactory methods were refactored to take advantage of this Builder
- Starting small here - rather than reconfigure the entire TestDataFactory in a single PR, just changing a few methods at a time. As we move forward and write more tests, we can continue to migrate the creation of some objects to the TestDataBuilder, especially in any tests that don't require a backing database.

## Additional information

Open to feedback on what would be most useful to tackle next with this refactor - my guess would be device and facility creation, but are there any particular tests that would benefit from this reworking? I'd rather start with the affected tests and move back towards the refactor.

## Testing

- All tests pass

## Checklist for Primary Reviewer
- [x] Changes comply with the SimpleReport Style Guide

### CODE REVIEW

**Review by zdeveloper — APPROVED** [review]
LGTM! exciting to finally see this!

File: backend/src/test/java/gov/cdc/usds/simplereport/test_util/TestDataFactory.java [hunk 1]
```diff
@@ -83,74 +82,70 @@ public class TestDataFactory {
...
+  @Autowired private DeviceSpecimenTypeRepository deviceSpecimenTypeRepository;
+  @Autowired private SupportedDiseaseRepository supportedDiseaseRepository;
+  @Autowired private ResultRepository resultRepository;
+  @Autowired private DemoOktaRepository oktaRepository;
+  @Autowired private TestResultUploadRepository testResultUploadRepository;
+
+  @Autowired private ApiUserService apiUserService;
+  @Autowired private DiseaseService diseaseService;
+
+  //  @Autowired private TestDataBuilder dataBuilder;
```
Comment (BobanL) [inline backend/src/test/java/gov/cdc/usds/simplereport/test_util/TestDataFactory.java:107]: In your next pr can you remove this comment?

**Review by BobanL — APPROVED** [review]
one little thing to fix in your next pr, but all looks good! Thank you!!!

File: backend/src/test/java/gov/cdc/usds/simplereport/test_util/TestDataFactory.java (hunk 1 above)
Comment (emmastephenson) [reply backend/src/test/java/gov/cdc/usds/simplereport/test_util/TestDataFactory.java]: Oh good call, just removed it! (Didn't particularly want to push the mistake to `main`)

**Review by BobanL — APPROVED** [review]

**Review by zdeveloper — APPROVED** [review]

**Pull request merged**